pywbem/_subscription_manager.py

```diff
@@ -1162,26 +1170,28 @@ def _create_destination(self, server_id, dest_url, owned):
         # Validate server_id
         server = self._get_server(server_id)
 
+        svr_name = self._get_systemname(server_id)
+
         # Validate the URL by reconstructing it. Do not allow defaults
         _, _, listener_url = parse_url(dest_url, allow_defaults=False)
 
-        this_host = getfqdn()
         ownership = "owned" if owned else "permanent"
 
-        dest_path = CIMInstanceName(DESTINATION_CLASSNAME,
-                                    namespace=server.interop_ns)
-
         dest_inst = CIMInstance(DESTINATION_CLASSNAME)
-        dest_inst.path = dest_path
         dest_inst['CreationClassName'] = DESTINATION_CLASSNAME
         dest_inst['SystemCreationClassName'] = SYSTEM_CREATION_CLASSNAME
-        dest_inst['SystemName'] = this_host
+        # Issue #2704, dummy set because mock requires the property
+        dest_inst['SystemName'] = svr_name or "dummy"
```

Comment:
If `_get_systemname()` is changed as suggested, its return value will never be `None`.

Reply:
Removed the method completely.  Since we put value in self._systemname when we add_server, this method no longer necessary.

DONE